aws/resource_aws_cognito_user_pool_domain.go

```diff
@@ -0,0 +1,170 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoUserPoolDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoUserPoolDomainCreate,
+		Read:   resourceAwsCognitoUserPoolDomainRead,
+		Delete: resourceAwsCognitoUserPoolDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
What do you think of adding validation for the domain also, as in http://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateUserPoolDomain.html#API_CreateUserPoolDomain_RequestSyntax?